### DISCUSSION TITLE
Why can't air be used for hot deployment in production environments? What are the differences between air and tableflip?

### DISCUSSION THREAD

**Discussion by plutoyty** [air-verse/air#471] [General]
Why can't air be used for hot deployment in production environments? What are the differences between air and tableflip?

**Comment by xiantang** [comment]
1. because air is still not stable enough
2. air aims to restart the server when the source code is changed, but as far as I know that We often run binary to start a go service
3. air will cause downtime, because restart service will not reuse old sockets allocated in the old process. but tablefilp supports it.
https://pkg.go.dev/github.com/cloudflare/tableflip#pkg-overview
